### PR TITLE
[Page Header]Implement new page header for admin data sources page

### PR DESCRIPTION
**Pull request by weiwang118**
### Description
Implement new page header and Look&Feel UI changes for Admin->Data Sources page. 

-  In data sources home page, change connection type tabs to button groups, and mount it on the top controls.
-  Mount create data source button and other action buttons on the top.
-  Remove duplicate head titles.
-  For connect s3 and connect prometheus page, change to one step page instead of previous two steps(Configure/Review) pages.
- Make search bar, button, title to use small or compressed property.

## Screenshot
New Page
Screenshot 2024-08-23 at 15 44 07
Screenshot 2024-08-23 at 15 44 47
Screenshot 2024-08-23 at 15 45 08
Screenshot 2024-08-23 at 15 46 02
Screenshot 2024-08-23 at 15 46 23
Screenshot 2024-08-23 at 15 46 39
Screenshot 2024-08-23 at 15 46 58
Screenshot 2024-08-23 at 15 47 19

## Changelog

- skip

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by kamingleung** [comment]
@weiwang118 Thank you for your updates, a few follow-up items:

1. For the connection type tabs, can we use the compressed button groups like in the mockups?
image

2. The top padding on the container should be the same length as the left, right padding.
image

3. For the Connect OpenSearch Cluster page, should we rename the action button to "Connect to OpenSearch cluster"? This will make it consistent with other create pages.
image

cc: @kgcreative @virajsanghvi

**Review by zhongnansu — COMMENTED** [review]
left 1 comment. Could you also check the failed CI?

File: src/plugins/data_source_management/public/components/create_data_source_wizard/components/create_form/create_data_source_form.tsx [hunk 1]
```diff
@@ -367,21 +372,41 @@ export class CreateDataSourceForm extends React.Component<
...
   };
 
   /* Render Section header*/
   renderSectionHeader = (i18nId: string, defaultMessage: string) => {
     return (
       <>
-        <EuiText grow={false}>
-          <h4>
+        <EuiText grow={false} size="s">
+          <h3>
```
[the remark this answers is not in the document]
Comment (zhongnansu) [reply src/plugins/data_source_management/public/components/create_data_source_wizard/components/create_form/create_data_source_form.tsx:407]: This is by requirements of fit and finish. It applies to all plugins. If you think there are other pages that violates this, let's fix those too. I'll share with you the fit and finish guidance offline

[the remark this answers is not in the document]
Comment (weiwang118) [reply src/plugins/data_source_management/public/components/create_data_source_wizard/components/create_form/create_data_source_form.tsx]: Change section tile to h2 in new commit. kindly request approval. Thanks!!!
Screenshot 2024-08-27 at 14 01 37
Screenshot 2024-08-27 at 14 01 02
Screenshot 2024-08-27 at 14 01 15
Screenshot 2024-08-27 at 14 01 23

**Review by zhongnansu — APPROVED** [review]
Approve, assuming the CI can pass, be sure to also check the codecov CI to make sure we don't decrease the code coverage.

**Review by BionIT — APPROVED** [review]

**Comment by BionIT** [comment]
Looks like failed coverage check

**Pull request merged**